src/app/systelab-components/datepicker/datepicker.component.ts

```diff
@@ -162,78 +147,93 @@ export class Datepicker implements OnInit, AfterViewInit, DoCheck, OnDestroy {
 	}
 
 	public changeDate(): void {
-		let emit = true;
-		const today: Date = new Date();
-
 		if (this.currentCalendar && this.currentCalendar.inputfieldViewChild.nativeElement.value !== undefined) {
-
-			let dateStr: string = this.currentCalendar.inputfieldViewChild.nativeElement.value.trim();
-			dateStr = dateStr.toLowerCase();
+			const today = new Date();
+			const dateStr = this.currentCalendar.inputfieldViewChild.nativeElement.value.trim().toLowerCase();
 
 			if (dateStr.length >= 2) {
+				if (dateStr.toUpperCase().includes('D')) {
```

Comment:
I think that includes is not supported in IE: https://stackoverflow.com/questions/36574351/includes-not-working-in-internet-explorer, https://stackoverflow.com/questions/44315902/javascript-includes-not-working-in-ie/44315933

Reply:
Due to the fact that the getAmount function controls the endsWith, I think that the same condition can be used in the ChangeDate function